app/uk/gov/hmrc/eventhub/service/PublisherService.scala

```diff
@@ -62,7 +75,9 @@ class PublisherService @Inject() (
     mongoSetup.topics.find(_.name == topic) match {
       case None                                     => Left(NoEventTopic("No such topic"))
       case Some(topic) if topic.subscribers.isEmpty => Left(NoSubscribersForTopic("No subscribers for topic"))
-      case Some(_)                                  => Right(publish(event, subscriberRepos(topic)))
+      case Some(topic) if !matchingPath(event, topic.subscribers) =>
+        Left(NoMatchingConfigurationPath("The payload does not match any path defined in config"))
+      case Some(_) => Right(publish(event, subscriberRepos(topic)))
```

Comment:
See my comment above - which means all subscribers get the event if the first subscriber in the list has a non empty path that matches?